src/algorithms/algorithm.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <string>
+
+class Game;
+
+class Algorithm{
+private:
+    std::string name;
+public:
+    Algorithm(std::string name);
+    std::string get_name();
+    virtual void run(Game& game) = 0;
```

Comment:
One of the discussions about the model class addition:

This method should receive model reference as parameter